internal/cryptsetup/cryptsetup.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package cryptsetup
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+// Device is a LUKS device.
+type Device struct {
+	devicePath string
+	headerPath string
+	keyPath    string
+}
+
+// NewDevice creates a new Device instance. It doesn't interact with the device itself.
+func NewDevice(devicePath, headerPath, keyPath string) *Device {
+	return &Device{
+		devicePath: devicePath,
+		headerPath: headerPath,
+		keyPath:    keyPath,
+	}
+}
+
+// IsLuks wraps the cryptsetup isLuks command and returns a bool reflecting if the device is formatted as LUKS.
+func (d *Device) IsLuks(ctx context.Context) (bool, error) {
+	var exitErr *exec.ExitError
+	out, err := runCryptsetupCmd(ctx, "isLuks", "--verbose", d.devicePath)
+	if errors.As(err, &exitErr) && exitErr.ExitCode() == 1 {
+		// isLuks exits 1 if the device is not a LUKS device.
+		return false, nil
+	} else if errors.As(err, &exitErr) {
+		return false, fmt.Errorf("checking if device %s is LUKS encrypted: %w, stdout: %s, stderr: %s", d.devicePath, err, out, exitErr.Stderr)
+	} else if err != nil {
+		return false, fmt.Errorf("checking if device %s is LUKS encrypted: %w", d.devicePath, err)
+	}
+	return true, nil
+}
+
+// Format wraps the luksFormat command.
+func (d *Device) Format(ctx context.Context) error {
+	if err := ensureBase(d.headerPath); err != nil {
+		return err
+	}
+	args := []string{
+		"luksFormat",
+		"--type=luks2",                          // Use LUKS2 header format.
+		"--cipher=aes-xts-plain64",              // Use AES-XTS cipher.
+		"--pbkdf=argon2id",                      // Use Argon2id as the key derivation function.
+		"--pbkdf-memory=10240",                  // Memory usage for Argon2i, limit to 10 MiB so it won't fail in low-memory pods.
+		"--batch-mode",                          // Suppresses all confirmation questions.
+		fmt.Sprintf("--key-file=%s", d.keyPath), // Path to the key file.
+		d.devicePath,
```

Comment:
Since the parameters are fixed, we should be able to tell whether this is compatible or not with prior cryptsetup volumes (referring to the "might be breaking" in the PR description).